cmd/git/git.go

```diff
@@ -1026,167 +1028,76 @@ func (j *DSGit) SetParentCommitFlag(richItem map[string]interface{}) (err error)
+		commit.CommittedTimestamp, _ = doc["commit_date"].(time.Time)
 		createdOn := authoredDt
-		commitRoles := []*models.CommitRole{}
+		commit.Orphaned = doc["orphaned"].(bool)
+		commit.SyncTimestamp = time.Now()
+		commitRoles := []insights.Contributor{}
 		identsAry, okIdents := doc["idents"].([][3]string)
 		identTypesAry, okIdentTypes := doc["ident_types"].([]string)
 		if okIdents && okIdentTypes {
 			for i := range identTypesAry {
-				var (
-					dt     time.Time
-					dtInTz time.Time
-					tz     float64
-				)
+				commitRole := insights.Contributor{}
 				ident := identsAry[i]
 				identType := identTypesAry[i]
-				if identType == "committer" {
-					dt = committedDt
-					dtInTz = committedInTz
-					tz = committedTz
-				} else {
-					dt = authoredDt
-					dtInTz = authoredInTz
-					tz = authoredTz
-				}
+				commitRole.Role = identType
+				commitRole.Weight = 1.0
 				name := ident[0]
 				username := ""
 				email := ident[2]
 				name, username = shared.PostprocessNameUsername(name, username, email)
-				userUUID := shared.UUIDAffs(ctx, source, email, name, username)
-				identity := &models.Identity{
-					ID:           userUUID,
-					DataSourceID: source,
-					Name:         name,
-					Username:     username,
+				nameList := strings.Split(name, "")
+				commitRole.Identity = user.UserIdentity{
+					Email:      email,
+					FirstName:  nameList[0],
+					IsVerified: false,
+					Username:   username,
+					Source:     source.Slug,
 				}
-				commitRoleUUID := shared.UUIDNonEmpty(ctx, docUUID, fmt.Sprintf("%s-%d", identType, i), userUUID)
-				commitRole := &models.CommitRole{
-					ID:       commitRoleUUID,
-					Name:     identType,
-					User:     identity,
-					Weight:   1.0,
-					When:     strfmt.DateTime(dt),
-					WhenInTz: strfmt.DateTime(dtInTz),
-					WhenTz:   tz,
+				if len(nameList) > 1 {
+					commitRole.Identity.LastName = strings.Join(nameList[1:], " ")
 				}
 				commitRoles = append(commitRoles, commitRole)
 			}
 		}
-		files := []*models.CommitFileAction{}
+		commit.Contributors = commitRoles
+		files := []insights.CommitFilesByType{}
 		fileAry, okFileAry := doc["file_data"].([]map[string]interface{})
 		if okFileAry {
 			for _, fileData := range fileAry {
-				var (
-					pAdded   *int64
-					pRemoved *int64
-					pChanged *int64
-				)
-				action, _ := fileData["action"].(string)
-				name, _ := fileData["name"].(string)
-				iAdded, _ := fileData["added"].(int)
-				iRemoved, _ := fileData["removed"].(int)
-				added := int64(iAdded)
-				removed := int64(iRemoved)
-				if added > 0 {
-					pAdded = &added
-				}
-				if removed > 0 {
-					pRemoved = &removed
-				}
-				changed := added + removed
-				if changed > 0 {
-					pChanged = &changed
-				}
-				file := &models.CommitFileAction{
-					Action:  action,
-					Name:    name,
-					Added:   pAdded,
-					Removed: pRemoved,
-					Changed: pChanged,
-				}
+				// TODO need to add logic to separate by the file type
```

Comment:
I think we should just list all files (including file name) not aggregate by anything.

Reply:
done.